src/mobile/MobileToolbarBack.js

```diff
@@ -6,21 +6,24 @@ export default class MobileToolbarBack extends Component {
 
 	static propTypes = {
 		backBtnBarTitle: PropTypes.string,
+		actions: PropTypes.object,
+		to: PropTypes.string,
 	};
-
+	goBack(e) {
+		const { actions } = this.props;
+		e.preventDefault();
+		actions.resetAssetPickerFilter();
```

Comment:
That is really not correct.
You should not have to change this file at all.
resetAssetPickerFilter does not belong here.